File: domain/src/main/java/com/peratrack/domain/models/Product.java
package com.peratrack.domain.models;

public class Product {
    public String name;
    public Float count;
    public CountType countType;
    public Float amount;
    public String currency;

    public enum CountType {
        KILOGRAMS("kilograms"),
        UNITS("units"),
        GRAMS("grams");

        private final String displayName;

        CountType(String displayName) {
            this.displayName = displayName;
        }

        @Override
        public String toString() {
            return displayName;
        }
    }

    public Product(
            String name,
            Float count,
            CountType countType,
            Float amount,
            String currency
    ) {
        this.name = name;
        this.count = count;
        this.countType = countType;
        this.amount = amount;
        this.currency = currency;
    }

    @Override
    public String toString() {
        if (name != null && count != null && countType != null
                && amount != null && currency != null) {
            return "{" + name + " " + count + " " +
                    countType + " " + amount + " " + currency + "} ";
        }
        return super.toString();
    }
}

File: data/src/main/java/com/peratrack/data/room/models/Product.kt
package com.peratrack.data.room.models

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.ForeignKey

@Entity(
    tableName = "product",
    foreignKeys = [ForeignKey(
        entity = Receipt::class,
        parentColumns = ["uid"],
        childColumns = ["receiptId"],
        onDelete = ForeignKey.CASCADE
    )]
)
data class Product(
    @PrimaryKey(autoGenerate = true) val productId: Int = 0,
    val receiptId: Int,  // Связь с Receipt
    val name: String,
    val count: Float,
    val countType: CountType,
    val amount: Float,
    val currency: String
) {
    enum class CountType {
        KILOGRAMS, UNITS, GRAMS
    }
}